filters/OverlayFilter.cpp

```diff
@@ -175,13 +186,30 @@ bool OverlayFilter::processOne(PointRef& point)
 
 void OverlayFilter::filter(PointView& view)
 {
-    PointRef point(view, 0);
-    for (PointId id = 0; id < view.size(); ++id)
+    point_count_t npoints = view.size();
+    unsigned int chunk_size = npoints / m_threads;
+    if (npoints % m_threads) chunk_size++;
+    std::vector<std::thread> threadList(m_threads);
+
+    for (int t = 0; t < m_threads; t++)
     {
-        point.setPointId(id);
-        processOne(point);
+        threadList[t] = std::thread(std::bind(
```

Comment:
My C++ knowledge is limited, but is a `std::bind` really necessary here?

Reply:
@abellgithub Would you mind sharing your expert view on this?

Reply:
The `std::bind` is not necessary. The arguments are forwarded to the thread when started. There are a bunch of ways this could be written that would be just fine but I think what's there is fine and would be better if the unnecessary `bind` is removed.